pkg/utils/apply/apply.go

```diff
@@ -107,13 +107,39 @@ func loggingApply(msg string, desired client.Object) {
 	klog.InfoS(msg, "name", d.GetName(), "resource", desired.GetObjectKind().GroupVersionKind().String())
 }
 
+// filterRecordForSpecial will filter special object that can reduce the record for "app.oam.dev/last-applied-configuration" annotation.
+func filterRecordForSpecial(desired client.Object) bool {
+	gvk := desired.GetObjectKind().GroupVersionKind()
+	gp, kd := gvk.Group, gvk.Kind
+	if gp == "" {
+		// group is empty means it's Kubernetes core API, we won't record annotation for Secret and Configmap
+		if kd == "Secret" || kd == "ConfigMap" {
+			return false
+		}
+		if _, ok := desired.(*corev1.ConfigMap); ok {
+			return false
+		}
+		if _, ok := desired.(*corev1.Secret); ok {
+			return false
+		}
+	}
+	ann := desired.GetAnnotations()
+	if ann != nil {
+		lac := ann[oam.AnnotationLastAppliedConfig]
+		if lac == "-" || lac == "skip" {
+			return false
+		}
+	}
+	return true
+}
+
 // Apply applies new state to an object or create it if not exist
 func (a *APIApplicator) Apply(ctx context.Context, desired client.Object, ao ...ApplyOption) error {
 	_, err := generateRenderHash(desired)
 	if err != nil {
 		return err
 	}
-	applyAct := &applyAction{updateAnnotation: true}
+	applyAct := &applyAction{updateAnnotation: filterRecordForSpecial(desired)}
```

Comment:
There is an ApplyOption with the name DisableUpdateAnnotation, can we use this option?

Reply:
I know that option, but this is an initial and default behavior in this case